### PR TITLE
Make the Chimple app bilingual

### PR DESCRIPTION
**Pull request by vishalmry**
Issue #59

### CODE REVIEW

File: goa/Classes/menu/MainMenuHome.cpp [hunk 1]
```diff
@@ -194,15 +194,36 @@ void MainMenuHome::bindEvents(cocos2d::Node *rootNode)
                 if (textTitle)
                 {
                     std::string translatedString = this->currentLangUtil->translateString(textTitle->getName());
-                    textTitle->setText(translatedString);
+
```
Comment (ankitk20) [inline goa/Classes/menu/MainMenuHome.cpp]: Avoid blank lines without any reason

File: goa/Classes/menu/MapScene.cpp [hunk 2]
```diff
@@ -159,31 +159,46 @@ void MapScene::processChildNodes(cocos2d::Node *rootNode) {
...
+                        englishName = "Dar es Salaam";
                     } else if(lName.compare("Camp") == 0){
 						lName = "·ñ¢UUUÂ";//Camp
+                        englishName = "Camp";
 					} else if(lName.compare("Farm House") == 0){
 						lName ="ÈUUUæ×ü ãæ©â";//Farm House
+                        englishName = "Farm House";
 					} else if(lName.compare("Mining Town") == 0){
 						lName ="¹ÙÙ àæãÚ";//Mining Town
+                        englishName = "Mining Town";
```
Comment (ankitk20) [inline goa/Classes/menu/MapScene.cpp:185]: Please format it properly

File: goa/Classes/menu/MapScene.cpp [hunk 3]
```diff
@@ -159,31 +159,46 @@ void MapScene::processChildNodes(cocos2d::Node *rootNode) {
...
                     std::string mapText = this->currentLangUtil->translateString(lName);
                     
-                    cocos2d::Label* label = Label::createWithTTF(mapText,"fonts/Chanakya.ttf", 150);
-                    label->setPosition(Vec2(node->getPosition().x, node->getPosition().y));
+                    cocos2d::Label* label = Label::createWithTTF(mapText,"fonts/Chanakya.ttf", 100);
+                    label->setPosition(Vec2(node->getPosition().x, node->getPosition().y - 50));
                     label->setColor(Color3B(128, 64, 0));
                     mainLayer->addChild(label);
+
+                    // Add English Label above the hindi text
```
Comment (ankitk20) [inline goa/Classes/menu/MapScene.cpp:201]: Make it "**a**rial.ttf"

File: goa/Classes/menu/ScrollableGameMapScene.cpp [hunk 4]
```diff
@@ -411,11 +415,25 @@ cocos2d::ui::Button* ScrollableGameMapScene::createButton(const rapidjson::Value
             button->loadTextureDisabled(buttonNormalIcon);
             button->addTouchEventListener(CC_CALLBACK_2(ScrollableGameMapScene::disabledGameSelected, this));
         }
-        
+
+        string gameTitle = gameJson["title"].GetString();
+        string delimiter = "$#$";
```
Comment (ankitk20) [inline goa/Classes/menu/ScrollableGameMapScene.cpp]: Shouldn't this be a const static variable?

File: goa/Classes/menu/ScrollableGameMapScene.cpp [hunk 5]
```diff
@@ -411,11 +415,25 @@ cocos2d::ui::Button* ScrollableGameMapScene::createButton(const rapidjson::Value
...
+            gameTitleEnglish += "/" + gameTitleHindi;
+        }
+
+        // IF THE LAST ROW IS NOT ENABLED
         button->setName(gameJson["name"].GetString());
-        button->setTitleText(LangUtil::getInstance()->translateString(gameJson["title"].GetString()));
+        button->setTitleText(LangUtil::getInstance()->translateString(gameTitleEnglish));
         button->setTitleAlignment(TextHAlignment::CENTER, TextVAlignment::BOTTOM);
-        button->setTitleFontName("fonts/Chanakya.ttf");
+        // button->setTitleFontName("fonts/Arial.ttf");
```
Comment (ankitk20) [inline goa/Classes/menu/ScrollableGameMapScene.cpp]: Remove this line if not required

File: goa/Classes/menu/ScrollableGameMapScene.cpp [hunk 6]
```diff
@@ -424,10 +442,18 @@ cocos2d::ui::Button* ScrollableGameMapScene::createButton(const rapidjson::Value
...
         }
         button->setTitleColor(titleColor);
-        button->setTitleFontSize(150);
+        button->setTitleFontSize(100);
         auto label = button->getTitleRenderer();
         label->setPosition(Vec2(label->getPositionX(), label->getPositionY()- 300));
         button->setScale(0.5);
+        
+        if (!isLastRow)
+        {
```
Comment (ankitk20) [inline goa/Classes/menu/ScrollableGameMapScene.cpp]: Add a comment here for the if condition

File: goa/proj.android-studio/goa/src/main/res/values/strings.xml [hunk 7]
```diff
@@ -2,6 +2,7 @@
 <resources>
     <string name="app_name">चिम्पल - Chimple</string>
     <string name="content_extraction">यह प्रक्रिया केवल एक बार की जाएगी। इसमें १० मिनट तक का समय लग सकता है।</string>
```
Comment (ankitk20) [inline goa/proj.android-studio/goa/src/main/res/values/strings.xml]: Make it hindi_content_extraction

File: goa/proj.android-studio/goa/src/main/res/values/strings.xml [hunk 8]
```diff
@@ -2,6 +2,7 @@
...
+    <string name="english_content_extraction">This is a one time process, It may take up to 10 minutes to complete.</string>
     <string name="bali_install_message">चिम्पल के लिए बाली एप डाउनलोड करे।</string>
```
Comment (ankitk20) [inline goa/proj.android-studio/goa/src/main/res/values/strings.xml]: Remove bali_install_message string

**Review by ankitk20 — CHANGES_REQUESTED** [review]
Update the code, recompile, test and then update this PR

File: goa/Classes/menu/MapScene.cpp (hunk 2 above)
Comment (vishalmry) [reply goa/Classes/menu/MapScene.cpp:185]: It is formatted in the text editor. Github is showing it like this.

File: goa/Classes/menu/MapScene.cpp (hunk 3 above)
Comment (vishalmry) [reply goa/Classes/menu/MapScene.cpp:201]: The mapText variable has the text in the Chanakya encoded form.

**Review by ankitk20 — DISMISSED** [review]

**Review by ankitk20 — DISMISSED** [review]

File: goa/Classes/menu/MainMenuHome.cpp [hunk 9]
```diff
@@ -194,15 +194,35 @@ void MainMenuHome::bindEvents(cocos2d::Node *rootNode)
...
+                        mainMenuEnglishText = "Map";
+
+                    textTitle->setText(mainMenuEnglishText);
                     textTitle->setTextHorizontalAlignment(TextHAlignment::CENTER);
                     textTitle->setTextVerticalAlignment(TextVAlignment::TOP);
                     textTitle->setEnabled(false);
                     textTitle->setTouchEnabled(false);
                     textTitle->setFocusEnabled(false);
-                    textTitle->setFontName("fonts/Chanakya.ttf");
+                    textTitle->setFontName("Arial");
```
Comment (mathewnik90) [inline goa/Classes/menu/MainMenuHome.cpp]: Is this the same font from the parent repository?

File: goa/Classes/menu/ScrollableGameMapScene.cpp [hunk 10]
```diff
@@ -316,6 +318,9 @@ bool ScrollableGameMapScene::init(std::string subGameMenuName) {
             for (int i = 0; i < numRows; i++) {
                 for (int j = 0; j < numCols; j++) {
                     if(index < orderedGameIndexes.size()) {
+                        isLastRow = false;
+                        if (i == numRows - 1)
```
Comment (mathewnik90) [inline goa/Classes/menu/ScrollableGameMapScene.cpp]: Use better variable names

**Review by mathewnik90 — DISMISSED** [review]

**Review by ankitk20 — DISMISSED** [review]

File: goa/Classes/menu/MainMenuHome.cpp [hunk 11]
```diff
@@ -194,15 +205,46 @@ void MainMenuHome::bindEvents(cocos2d::Node *rootNode)
                 if (textTitle)
                 {
                     std::string translatedString = this->currentLangUtil->translateString(textTitle->getName());
-                    textTitle->setText(translatedString);
+                    string gameName = textTitle->getName().c_str();
+                    string mainMenuEnglishText = "";
```
Comment (mathewnik90) [inline goa/Classes/menu/MainMenuHome.cpp:209]: Check if you can directly update the source

Comment (vishalmry) [reply goa/Classes/menu/MainMenuHome.cpp:209]: If we use the text from the source then the text is left aligned like below eg which inconsistent with other parts of app.
**English_text
Hindi_text**

**Review by ankitk20 — APPROVED** [review]

**Comment by ankitk20** [comment]
#59

**Pull request merged**